### PR TITLE
Add (public) sendPayload method to Usage

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

**Comment by kevmoo** [comment]
I've hit cases where I'd like to intercept *all* connects to the server. This makes it easy.

**Comment by devoncarew** [comment]
This interception currently works by assuming subclassing?

**Comment by kevmoo** [comment]
> This interception currently works by assuming subclassing?

Yup. Just like `pkg:http`, etc

**Comment by devoncarew** [comment]
sgtn. I don't think this is a breaking change then - we currently assume clients have not subclassed. I'd prefer not rev'ing the major version, since that has downstream costs.

**Comment by kevmoo** [comment]
@devoncarew – well, my next PR does change everything to `extends` – eliminates a lot of duplication between the impl and mock class

WDYT?

File: pubspec.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # BSD-style license that can be found in the LICENSE file.
 
 name: usage
-version: 3.4.0
+version: 4.0.0-dev
```
Comment (devoncarew) [inline pubspec.yaml]: As discussed, here and in the changelog, let's drop this back to `3.5.0-dev`.

**Comment by kevmoo** [comment]
Eh, going with something else...

**Pull request closed without merging**